File: java/03-bootstrap/src/main/java/com/fResult/bootstrap/customers/controllers/CustomerController.java
package com.fResult.bootstrap.customers.controllers;

import com.fResult.bootstrap.customers.Customer;
import com.fResult.bootstrap.customers.services.CustomerService;
import java.util.Collection;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/customers")
public class CustomerController {
  private final CustomerService customerService;

  public CustomerController(CustomerService customerService) {
    this.customerService = customerService;
  }

  @GetMapping
  public Collection<Customer> all() {
    return customerService.findAll();
  }

  @GetMapping("/{id}")
  public Customer byId(@PathVariable Long id) {
    return customerService.findById(id);
  }

  @PostMapping
  public Collection<Customer> create(@RequestBody Customer body) {
    return customerService.save(body.name());
  }
}

File: kotlin/03-bootstrap/src/main/kotlin/com/fResult/bootstrap/customers/controllers/CustomerController.kt
package com.fResult.bootstrap.customers.controllers

import com.fResult.bootstrap.customers.Customer
import com.fResult.bootstrap.customers.services.CustomerService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/customers")
class CustomerController(val customerService: CustomerService) {
  @GetMapping
  fun all() = customerService.findAll()

  @GetMapping("/{id}")
  fun byId(@PathVariable id: Long): Customer? = customerService.findById(id)

  @PostMapping
  fun create(@RequestBody body: Customer): Collection<Customer> = customerService.save(body.name)
}
